lib/metrics/context.js

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+var isA = require('joi')
+var HEX = require('../routes/validators').HEX_STRING
+
+module.exports = {
+  schema: isA.object({
+    flowId: isA.string().length(32).regex(HEX).required(),
+    flowBeginTime: isA.number().integer().positive().required(),
+    context: isA.string().optional(),
+    entrypoint: isA.string().optional(),
+    migration: isA.string().optional(),
+    service: isA.string().optional(),
+    utmCampaign: isA.string().optional(),
+    utmContent: isA.string().optional(),
+    utmMedium: isA.string().optional(),
+    utmSource: isA.string().optional(),
+    utmTerm: isA.string().optional()
+  }).optional(),
+
+  add: function (data, metadata) {
+    if (metadata) {
+      data.time = Date.now()
+      data.flow_id = metadata.flowId
+      data.flow_time = data.time - metadata.flowBeginTime
+      data.context = metadata.context
+      data.entrypoint = metadata.entrypoint
+      data.migration = metadata.migration
+      data.service = metadata.service
```

Comment:
`log.activityEvent` currently has some special logic to pull this out of top-level query params or request body; we'll need to ensure it doesn't overwrite any value specified here.